lib/python/qmk/commands.py

```diff
@@ -0,0 +1,57 @@
+"""Functions that build make commands
+"""
+import json
+import qmk.keymap
+
+def create_make_command(keyboard, keymap, target=None):
+    """Create a make compile command
+
+    Args:
+        keyboard
+            The path of the keyboard, for example 'plank'
+
+        keymap
+            The name of the keymap, for example 'algernon'
+
+        target
+            Usually a bootloader.
+
+    Returns:
+        A command that can be run to make the specified keyboard and keymap
+    """
+    if target is None:
+        return ['make', ':'.join((keyboard, keymap))]
```

Comment:
bear with me, since I have literally zero experience with python, but this may be better: 
```suggestion
        return ['make', ':'.join((keyboard, keymap)), ':flash']
```

This way, if no bootloader is specified, it runs the `:flash` target. But you can override by setting the target (in the case that the bootloader is wrong, eg pro micro vs elite c).

Reply:
Or you could just drop the `if` statement and write:
```suggestion
        return ['make', ':'.join((keyboard, keymap, target if target else "flash"))]
```

Reply:
Im a bit hesitant to make this assumption in `commands.py`, as this code is used by both `flash` and `compile`.

Reply:
Yeah, ideally, we want to have `flash` and `compile` managed separately.  There are a number of instances where you want the keyboard compiled, but you don't want to flash the board. 

A good, future example of this is if everything switches over to the qmk cli, we don't want travis to attempt to flash everything, since that would definitely fail. 
Also, I do a lot of testing, and I don't always want to flash, nor escape out of the flashing.

Reply:
Okay! Currently only `flash` makes the assumption to use `:flash` as default. So I will leave `commands.py` untouched.